ICE: 'compiler/rustc_resolve/src/lib.rs:1929:21: undetermined resolution during main resolution pass', /rustc/51748a8fc77283914d4135f31b5966a407208187/library/std/src/panic.rs:59:5

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.

http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/

-->

### Code

```Rust
use rustc_expand::base::SyntaxExtension;
fn register_builtin_macros() {
    macro register_derive ($($f : expr,)) { $ (Box :: SyntaxExtension($f)) }
    register_derive! { clone, }
}
```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.52.0-nightly (51748a8fc 2021-03-05)
binary: rustc
commit-hash: 51748a8fc77283914d4135f31b5966a407208187
commit-date: 2021-03-05
host: x86_64-unknown-linux-gnu
release: 1.52.0-nightly
LLVM version: 12.0.0
```

### Error output

```
error: expected one of: `*`, `+`, or `?`
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:3:29
  |
3 |     macro register_derive ($($f : expr,)) { $ (Box :: SyntaxExtension($f)) }
  |                             ^^^^^^^^^^^^

error: expected one of: `*`, `+`, or `?`
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:3:47
  |
3 |     macro register_derive ($($f : expr,)) { $ (Box :: SyntaxExtension($f)) }
  |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: maybe a missing crate `rustc_expand`?
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:1:5
  |
1 | use rustc_expand::base::SyntaxExtension;
  |     ^^^^^^^^^^^^ maybe a missing crate `rustc_expand`?

thread 'rustc' panicked at 'compiler/rustc_resolve/src/lib.rs:1929:21: undetermined resolution during main resolution pass', /rustc/51748a8fc77283914d4135f31b5966a407208187/library/std/src/panic.rs:59:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.52.0-nightly (51748a8fc 2021-03-05) running on x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0433`.
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
error: expected one of: `*`, `+`, or `?`
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:3:29
  |
3 |     macro register_derive ($($f : expr,)) { $ (Box :: SyntaxExtension($f)) }
  |                             ^^^^^^^^^^^^

error: expected one of: `*`, `+`, or `?`
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:3:47
  |
3 |     macro register_derive ($($f : expr,)) { $ (Box :: SyntaxExtension($f)) }
  |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: maybe a missing crate `rustc_expand`?
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:1:5
  |
1 | use rustc_expand::base::SyntaxExtension;
  |     ^^^^^^^^^^^^ maybe a missing crate `rustc_expand`?

thread 'rustc' panicked at 'compiler/rustc_resolve/src/lib.rs:1929:21: undetermined resolution during main resolution pass', /rustc/51748a8fc77283914d4135f31b5966a407208187/library/std/src/panic.rs:59:5
stack backtrace:
   0: std::panicking::begin_panic
   1: std::panic::panic_any
   2: rustc_middle::ty::context::tls::with_opt
   3: rustc_middle::util::bug::opt_span_bug_fmt
   4: rustc_middle::util::bug::span_bug_fmt
   5: rustc_resolve::Resolver::resolve_ident_in_lexical_scope
   6: rustc_resolve::Resolver::resolve_path_with_ribs::{{closure}}
   7: rustc_resolve::Resolver::resolve_path_with_ribs
   8: rustc_resolve::late::LateResolutionVisitor::resolve_qpath_anywhere
   9: rustc_resolve::late::LateResolutionVisitor::smart_resolve_path_fragment
  10: rustc_resolve::late::LateResolutionVisitor::resolve_expr
  11: rustc_resolve::late::LateResolutionVisitor::resolve_expr
  12: <rustc_resolve::late::LateResolutionVisitor as rustc_ast::visit::Visitor>::visit_block
  13: <rustc_resolve::late::LateResolutionVisitor as rustc_ast::visit::Visitor>::visit_fn
  14: rustc_ast::visit::walk_item
  15: rustc_resolve::late::LateResolutionVisitor::resolve_item
  16: rustc_resolve::late::<impl rustc_resolve::Resolver>::late_resolve_crate
  17: rustc_session::utils::<impl rustc_session::session::Session>::time
  18: rustc_resolve::Resolver::resolve_crate
  19: rustc_interface::passes::configure_and_expand_inner
  20: rustc_interface::passes::configure_and_expand::{{closure}}
  21: rustc_data_structures::box_region::PinnedGenerator<I,A,R>::new
  22: rustc_interface::passes::configure_and_expand
  23: rustc_interface::queries::Queries::expansion
  24: rustc_interface::queries::<impl rustc_interface::interface::Compiler>::enter
  25: rustc_span::with_source_map
  26: rustc_interface::interface::create_compiler_and_run
  27: rustc_span::with_session_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.52.0-nightly (51748a8fc 2021-03-05) running on x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0433`.
```

</p>
</details>

--- a/src/test/ui/resolve/issue-82865.rs
+++ b/src/test/ui/resolve/issue-82865.rs
@@ -1,0 +1,13 @@
+// Regression test for #82865.
+
+#![feature(decl_macro)]
+
+use x::y::z; //~ ERROR: failed to resolve: maybe a missing crate `x`?
+
+macro mac () {
+    Box::z //~ ERROR: no function or associated item
+}
+
+fn main() {
+    mac!();
+}

--- a/src/test/ui/resolve/issue-82865.stderr
+++ b/src/test/ui/resolve/issue-82865.stderr
@@ -1,0 +1,21 @@
+error[E0433]: failed to resolve: maybe a missing crate `x`?
+  --> $DIR/issue-82865.rs:5:5
+   |
+LL | use x::y::z;
+   |     ^ maybe a missing crate `x`?
+
+error[E0599]: no function or associated item named `z` found for struct `Box<_, _>` in the current scope
+  --> $DIR/issue-82865.rs:8:10
+   |
+LL |     Box::z
+   |          ^ function or associated item not found in `Box<_, _>`
+...
+LL |     mac!();
+   |     ------- in this macro invocation
+   |
+   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0433, E0599.
+For more information about an error, try `rustc --explain E0433`.

